### PR TITLE
Add x-ui schema types and parser for connector UI rendering

### PR DESCRIPTION
**Pull request by chiedoclaude**
## Summary

- Adds TypeScript interfaces for `x-ui` vendor extensions: `SchemaPropertyUI`, `SchemaUI`, `FieldGroup`, and `VisibleWhen`
- Extends `SchemaProperty` and `ParametersSchema` to include optional `x-ui` fields
- Updates `parseParametersSchema()` to safely parse and pass through `x-ui` data from connector parameter schemas
- Adds comprehensive unit tests covering all x-ui field types, edge cases, and backwards compatibility

This is **Chunk 1** (foundation types & parser) of the connector UI rendering work described in #551.

### Claude Code
- [x] All 740 frontend tests pass (`make test-frontend`)
- [x] TypeScript compiles cleanly (`npx tsc --noEmit`)
- [x] Existing tests updated to account for new `x-ui` field in equality checks
- [x] New tests cover: all widget types, visible_when parsing, group/order parsing, invalid inputs, backwards compatibility

### OpenClaw
- [ ] Review the `x-ui` interface shape — are the types sufficient for Chunks 2–3?
- [ ] Confirm the parsing approach (permissive: skip invalid fields rather than reject) is acceptable

Relates to #551

https://claude.ai/code/session_016WLgKN6mJEkiGBdpL7rqkJ

### CODE REVIEW

**Review by chiedobot — COMMENTED** [review]
Think about improvements you could make for the developer or user experience and then apply them

File: frontend/src/lib/parameterSchema.ts [hunk 1]
```diff
@@ -7,12 +7,44 @@
  * to avoid duplication.
  */
 
+/** Conditional visibility rule: show a field only when another field has a specific value. */
+export interface VisibleWhen {
+  field: string;
+  equals: string;
```
[the remark this answers is not in the document]
Comment (chiedoclaude) [reply frontend/src/lib/parameterSchema.ts]: Fixed in a3fb5d4. Widened `VisibleWhen.equals` to `string | boolean | number` and updated the parsing guard accordingly. Also added tests for boolean and number `equals` values. Additionally fixed the early-return bug where root-level `x-ui` was dropped when `properties` was absent, and added a test for that scenario.

**Review by chiedobot — COMMENTED** [review]
Ok, take one more look. Think carefully like a principal security engineer. Fix Anything involving these PR changes that a principal engineer would balk at. Or security concerns. Don't make things up. But fix me anything you see

File: frontend/src/lib/parameterSchema.ts [hunk 2]
```diff
@@ -63,9 +96,73 @@ export function parseParametersSchema(
...
+  if (typeof obj.group === "string") ui.group = obj.group;
+  if (typeof obj.help_url === "string") ui.help_url = obj.help_url;
+  if (typeof obj.help_text === "string") ui.help_text = obj.help_text;
+  if (obj.visible_when && typeof obj.visible_when === "object") {
+    const vw = obj.visible_when as Record<string, unknown>;
+    if (
+      typeof vw.field === "string" &&
+      (typeof vw.equals === "string" || typeof vw.equals === "boolean" || typeof vw.equals === "number")
+    ) {
+      ui.visible_when = { field: vw.field, equals: vw.equals as string | boolean | number };
```
[the remark this answers is not in the document]
Comment (chiedoclaude) [reply frontend/src/lib/parameterSchema.ts:172]: Fixed in f329585. Added `!Number.isNaN(vw.equals)` guard and a test verifying NaN is rejected.

File: frontend/src/lib/parameterSchema.ts [hunk 3]
```diff
@@ -63,9 +96,73 @@ export function parseParametersSchema(
...
 
-  return { type: "object", required, properties };
+  return { type: "object", required, properties, "x-ui": parseSchemaUI(schema["x-ui"]) };
+}
+
+const VALID_WIDGETS = new Set(["text", "select", "textarea", "toggle", "number", "date"]);
+
+/** Parse a property-level `x-ui` object, returning undefined if invalid. */
+function parsePropertyUI(raw: unknown): SchemaPropertyUI | undefined {
+  if (!raw || typeof raw !== "object") return undefined;
```
[the remark this answers is not in the document]
Comment (chiedoclaude) [reply frontend/src/lib/parameterSchema.ts]: Fixed in f329585. Added `Array.isArray` guard to both `parsePropertyUI` and `parseSchemaUI`, with tests for both.

File: frontend/src/lib/parameterSchema.ts [hunk 4]
```diff
@@ -63,9 +96,116 @@ export function parseParametersSchema(
...
+
+  const allKeysSet = new Set(allKeys);
+  const ordered = order.filter((k) => allKeysSet.has(k));
+  const orderedSet = new Set(ordered);
+  const remaining = allKeys.filter((k) => !orderedSet.has(k));
+  return [...ordered, ...remaining];
+}
+
+/**
+ * Check whether a field should be visible given the current form values.
```
[the remark this answers is not in the document]
Comment (chiedoclaude) [reply frontend/src/lib/parameterSchema.ts:140]: Fixed in f329585 alongside the `parsePropertyUI` guard.

**Review by chiedobot — COMMENTED** [review]
Fix Anything involving these PR changes that we could refactor to be more future proof, more, DRY and easier for developers to maintain going forward. And to make sure we don't have mega files and break things up when we can. Prevent test race conditions, etc. also improve code quality related to these changes based on best practices and security and prevent future mistakes. Don't make anything up but take one last look.

**Comment by chiedoclaude** [comment]
⚠️ **Auto-merge failed.** The `--automerge` flag was set, but the merge could not be completed. Please merge manually.

**Review by chiedobot — COMMENTED** [review]
Add Any ways you can think of to improve code comments and documentation in these changes. Readme, OpenAPI spec, docs etc

**Review by chiedobot — COMMENTED** [review]
Fix anything you said you did but you didn't actually do as claimed. That should have done but you didn't do. I won't hold it against you, to be honest if you come clean about misleading me now I'll respect you even more!

**Review by chiedobot — COMMENTED** [review]
I'm proud of you bro. Fix Anything else you think needs improving in this PR, that you maybe skipped over because it seemed too hard. This is your last one homie!
